src/meeting/MeetingContainer.js

```diff
@@ -17,15 +17,18 @@ import useMediaStream from "../hooks/useMediaStream";
 export function MeetingContainer({
   onMeetingLeave,
   setIsMeetingLeft,
+
   selectedMic,
   selectedWebcam,
-  selectWebcamDeviceId,
-  setSelectWebcamDeviceId,
-  selectMicDeviceId,
-  setSelectMicDeviceId,
-  micEnabled,
-  webcamEnabled,
+  selectedSpeaker,
+  setSelectedMic,
+  setSelectedWebcam,
+  micOn,
+  webcamOn,
+  isCameraPermissionAllowed,
```

Comment:
Can this states be moved to meetingAppProvider so that it does not need to be passed down in each component